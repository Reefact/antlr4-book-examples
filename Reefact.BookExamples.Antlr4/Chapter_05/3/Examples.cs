#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using NFluent;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_05._3 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void example_01() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("a[1]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree    = grun.ToLispStyleTree();
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(expr a [ (expr 1) ])");
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void example_02() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("(a[1])");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree    = grun.ToLispStyleTree();
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(expr ( (expr a [ (expr 1) ]) ))");
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void example_03() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("(((1)))");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string output = grun.ToLispStyleTree();
            // Verify
            Check.That(output).IsEqualTo("(expr ( (expr ( (expr ( (expr 1) )) )) ))");
        }

    }

}