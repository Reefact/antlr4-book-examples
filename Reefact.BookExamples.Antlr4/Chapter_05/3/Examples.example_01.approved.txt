graph TD
	1["expr"] --> 2["a"]
	1 --> 3["["]
	1 --> 4["expr"]
	4 --> 5["1"]
	1 --> 6["]"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;